pkg/http/api_metadata.go

```diff
@@ -0,0 +1,135 @@
+/*
+Copyright 2022 The Dapr Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package http
+
+import (
+	"bytes"
+	"encoding/json"
+
+	"github.com/valyala/fasthttp"
+	"google.golang.org/protobuf/encoding/protojson"
+	"google.golang.org/protobuf/types/known/emptypb"
+
+	runtimev1pb "github.com/dapr/dapr/pkg/proto/runtime/v1"
+)
+
+func (a *api) constructMetadataEndpoints() []Endpoint {
+	return []Endpoint{
+		{
+			Methods: []string{fasthttp.MethodGet},
+			Route:   "metadata",
+			Version: apiVersionV1,
+			Handler: a.onGetMetadata(),
+		},
+		{
+			Methods: []string{fasthttp.MethodPut},
+			Route:   "metadata/{key}",
+			Version: apiVersionV1,
+			Handler: a.onPutMetadata(),
+		},
+	}
+}
+
+func (a *api) onGetMetadata() fasthttp.RequestHandler {
+	return UniversalFastHTTPHandler(
+		a.universal.GetMetadata,
+		UniversalFastHTTPHandlerOpts[*emptypb.Empty, *runtimev1pb.GetMetadataResponse]{
+			OutModifier: func(out *runtimev1pb.GetMetadataResponse) (any, error) {
```

Comment:
I think it will be clearer to unmarshal the data into structs and then finally marshalling the struct into json. 
The part of modifying an already encoded json object is something that I think should be avoided.

Then if we are marshalling into different structs, should the metadata API be moved into universalAPI?